cowprotocol/cowswap
# feat(tx-bundling): enable bundled approval + presign for Safe App on Limit orders

# Summary

Same as #2434 but for Limit orders

## ⚠️ IMPORTANT ⚠️ 
* It only works for Safes - other wallets won't be supported for the time being
* It only works when using CoW Swap as a Safe App - WC support is pending on Safe implementation

# Testing

1. Load the app inside Safe
2. Go to limit orders tab
3. Pick as sell token one that has 0 approval
* Assuming there's enough liquidity, you should see `approve and swap`
<img width="484" alt="image" src="https://github.com/cowprotocol/cowswap/assets/43217/e6ca0cd0-56a4-4e15-a153-ee567c47008a">

4. Click on it
5. If not on expert mode, the confirmation modal should appear
* (Text hasn't changed yet, will be re-iterated later)
6. Click on it
* You should see the Safe tx pop up, with 2 actions: approve and swap
7. Submit the tx
* You should be prompted to sign it in your wallet
* Depending on your Safe setup, the tx will be executed or need more signatures
* Once the tx is executed, the order should change from `pre signing` to `open`

8. Now pick a sell token for which you have enough approval
* It should work as expected

9. Repeat steps `1` and `2` with a Safe connected via WC
* You should see the regular request for approval
10. Repeat steps `1` and `2` with an EOA
* You should see the regular request for approval
